LDAR_Sim/model_code/ldar_sim_main.py

```diff
@@ -24,47 +24,88 @@
-    program_list = ['P_ref', 'P_dev_continuous', 'P_cont']
-
-# -----------------------------Set up programs----------------------------------
-    programs = []
-
-    warnings.filterwarnings('ignore')    # Temporarily mute warnings
-
-    for p in range(len(program_list)):
-        file = wd + program_list[p] + '.txt'
-        exec(open(file).read())
-        programs.append(eval(program_list[p]))
-
-    n_processes = programs[0]['n_processes']
-    print_from_simulations = programs[0]['print_from_simulations']
-    n_simulations = programs[0]['n_simulations']
-    spin_up = programs[0]['spin_up']
-    ref_program = program_list[0]
-    write_data = programs[0]['write_data']
-
+    # Look for parameter files supplied as arguments - if parameter files are supplied as
+    # arguments, proceed to parse and type check input parameter type with the input manager
+    parameter_filenames = sys.argv[1:]
+
+    # FOR TESTING ONLY!
+    print('WARNING: USING HARD CODED INPUT FILES FOR TESTING ONLY - REMOVE ON LINE 43 OF ldar_sim_main.py')
+    parameter_filenames = [
+        '..//sample_simulations//simple_OGI.yaml',
+        '..//sample_simulations//P_ref.yaml',
+        '..//sample_simulations//OGI.yaml',
+    ]
+
+    if len(parameter_filenames) > 0:
+        print('LDAR-Sim using parameters supplied as arguments')
+        input_manager = InputManager()
+        simulation_parameters = input_manager.read_and_validate_parameters(parameter_filenames)
+
+        # Assign appropriate local variables to match older way of inputting parameters
+        wd = simulation_parameters['wd']
+        output_directory = simulation_parameters['output_directory']
+        programs = simulation_parameters['programs']
+        n_processes = simulation_parameters['n_processes']
+        print_from_simulations = simulation_parameters['print_from_simulations']
+        n_simulations = simulation_parameters['n_simulations']
+        spin_up = simulation_parameters['spin_up']
+        ref_program = simulation_parameters['reference_program']
+        write_data = simulation_parameters['write_data']
+        weather_file = simulation_parameters['weather_file']
+        start_year = simulation_parameters['start_year']
+
+    else:
+        print('LDAR-Sim using parameters coded into ldar_sim_main.py')
+        # ------------------------------------------------------------------------------
+        # -----------------------------Global parameters--------------------------------
+        src_dir_path = Path(os.path.dirname(os.path.realpath(__file__)))
+        src_dir = str(src_dir_path)
+        root_dir = str(src_dir_path.parent)
+        wd = os.path.abspath(root_dir) + "/inputs_template/"
+        output_directory = os.path.abspath(root_dir) + "/outputs/"
+        # Programs to compare; Position one should be the reference program (P_ref)
+        program_list = ['P_ref', 'P_dev_continuous', 'P_cont']
```

Comment:
We should choose what goes here. My vote is 'P_ref' (or some OGI program) and 'P_base'

Reply:
This can be anything - this is the way master was previously for the old 'P_*' style of running programs. I think after testing of method modules this will be tightened up? @KeeganShaw-GIS

Reply:
Yup, I agree with both, P_ref, P_base for testing just to avoid conflicts, but in the future this will be obsolete.

Reply:
Changed. Thanks.